File: MusicEventManagementSystem/MusicEventManagementSystem.Core/Models/DTOs/Client/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicEventManagementSystem.Core.Models.DTOs.Client
{
    public class OrderDto
    {
        public int OrderId { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public DateTime OrderDate { get; set; }
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = string.Empty;
        public string PaymentMethod { get; set; } = string.Empty;
        public int TotalTickets { get; set; }
        public List<OrderItemDto> Items { get; set; } = new();
    }

    public class OrderItemDto
    {
        public string? EventName { get; set; }
        public string? TicketTypeName { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal Subtotal { get; set; }
    }

    public class OrderTicketDto
    {
        public int TicketId { get; set; }
        public string? UniqueCode { get; set; }
        public string? QrCode { get; set; }
        public string? EventName { get; set; }
        public string? TicketTypeName { get; set; }
        public string? ZoneName { get; set; }
        public DateTime EventStartDate { get; set; }
        public decimal Price { get; set; }
        public string Status { get; set; } = string.Empty;
    }

    public class OrderDetailsDto
    {
        public int OrderId { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public DateTime OrderDate { get; set; }
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = string.Empty;
        public string PaymentMethod { get; set; } = string.Empty;
        public List<OrderTicketDto> Tickets { get; set; } = new();
        public List<AppliedOfferDto> AppliedOffers { get; set; } = new();
    }

    public class AppliedOfferDto
    {
        public string? OfferName { get; set; }
        public string? Description { get; set; }
        public decimal DiscountAmount { get; set; }
    }
}

File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/shared/types/api/order.ts
export interface OrderDto {
  orderId: number;
  orderNumber: string;
  orderDate: string;
  totalAmount: number;
  status: string;
  paymentMethod: string;
  totalTickets: number;
  items: OrderItemDto[];
  subtotal?: number;
  discount?: number;
  serviceFee?: number;
  totalItems?: number;
}

export interface OrderItemDto {
  eventName?: string;
  ticketTypeName?: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
}

export interface OrderTicketDto {
  ticketId: number;
  uniqueCode?: string;
  qrCode?: string;
  eventName?: string;
  ticketTypeName?: string;
  zoneName?: string;
  eventStartDate: string;
  price: number;
  status: string;
  eventDate?: string;
  venueName?: string;
  seatNumber?: string;
}

export interface OrderDetailsDto {
  orderId: number;
  orderNumber: string;
  orderDate: string;
  totalAmount: number;
  status: string;
  paymentMethod: string;
  tickets: OrderTicketDto[];
  appliedOffers: AppliedOfferDto[];
  subtotal?: number;
  discount?: number;
  serviceFee?: number;
  billingInfo?: BillingInfoDto;
}

export interface AppliedOfferDto {
  offerName?: string;
  description?: string;
  discountAmount: number;
}

export interface BillingInfoDto {
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  address?: string;
  city?: string;
  postalCode?: string;
  country?: string;
}